Implementation/Composing/index-v9.md

```diff
@@ -0,0 +1,519 @@
+| PropertyValueConverters   | Ordered   | Type scanned for `IPropertyValueConverter`                        |
+| SearchableTrees           | Lazy      | Type scanned for `ISearchableTree`                                |
+| Sections                  | Ordered   | Package.manifest & Explicit Registration                          |
+| TourFilters               | Base      | Empty collection                                                  |
+| Trees                     | Base      | Type scanned. Must inherit `TreeControllerBase` & use `[Tree]`    |
+| UrlProviders              | Ordered   | Explicit Registration                                             |
+| UrlSegmentProviders       | Ordered   | Explicit Registration                                             |
+| Validators                | Lazy      | Explicit Registration                                             |
+
+
+### Types of Collections
+
+Set<br/>
+`SetCollectionBuilderBase` - The base class for collection builders that do not order their items explicitly.
+
+Ordered<br/>
+`OrderedCollectionBuilderBase` - The base class for collection builders that order their items explicitly.
+
+Weighted<br/>
+`WeightedCollectionBuilder` - The base class for collection builders that order their items by the `[Weight]` attribute.
+
+Lazy<br/>
+`LazyCollectionBuilderBase` - The base class for collection builders that resolve the types at the last moment, only when the collection is required.
+
+
+### Example - Modifying Collections
+
+This example shows how to control which Healthchecks are available to run in the Umbraco backoffice. Create a C# class which implements IUserComposer, the Compose method gives access to the HealthChecks collection of the Umbraco Composition - first we clear all HealthChecks from the collection, then add back in the ones we want to keep:
+
+```csharp
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.HealthChecks.Checks.Permissions;
+using Umbraco.Cms.Core.HealthChecks.Checks.Security;
+
+namespace My.Website
+{
+    public class MyComposer: IComposer
+    {
+        public void Compose(IUmbracoBuilder builder)
+        {
+            // Remove all HealthChecks
+            builder.HealthChecks().Clear();
+
+            // Explicitly add back the ones we want to use
+            builder.HealthChecks().Add<FolderAndFilePermissionsCheck>();
+            builder.HealthChecks().Add<ExcessiveHeadersCheck>();
+        }
+    }
+}
+```
+
+
+## Attributes
+Umbraco has some useful C# attributes to decorate your composer classes or Types used in collections, to give you further control on how and when your Composers will 'compose'.
+
+### `[ComposeBefore]` and `[ComposeAfter]`
+A finer-grain mechanism can then be used to refine the order of composition. Each composer can specify that it should compose before or after another composer, using the ComposeBefore and ComposeAfter attributes. For instance:
```

Comment:
Hopefully we can nuke this section, perhaps with a section to talk about alternative approaches